GoMeet/src/test/java/com/google/sps/VoteMeetingTimeServletTest.java

```diff
@@ -0,0 +1,258 @@
+    when(mockedRequest.getParameter(MeetingTimeFields.VOTERS)).thenReturn(null); // null voter
+    new VoteMeetingTimeServlet().doPost(mockedRequest, mockedResponse);
+
+    // check that error response was sent with appropriate details
+    testBadRequest(HttpServletResponse.SC_BAD_REQUEST, ErrorMessages.BAD_REQUEST_ERROR);
+  }
+
+  @Test
+  public void testNoEntitiesFound() throws IOException {
+    // Create a datastore entity, then delete. Trying to retrieve using this entity key would
+    // return no results.
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Entity fakeMeetingTime = new Entity("MeetingTime");
+    datastore.put(fakeMeetingTime);
+    Key fakeMeetingTimeKey = fakeMeetingTime.getKey();
+    String fakeMeetingTimeKeyStr = KeyFactory.keyToString(fakeMeetingTimeKey);
+    datastore.delete(fakeMeetingTimeKey);
+
+    when(mockedRequest.getParameter(MeetingTimeFields.MEETING_TIME_ID)).thenReturn(fakeMeetingTimeKeyStr);
+    when(mockedRequest.getParameter(MeetingTimeFields.VOTERS)).thenReturn(NEW_VOTER);
+    new VoteMeetingTimeServlet().doPost(mockedRequest, mockedResponse);
+
+    // check that error response was sent with appropriate details
+    testBadRequest(HttpServletResponse.SC_NOT_FOUND, ErrorMessages.ENTITY_NOT_FOUND_ERROR);
+  }
+
+  @Test
+  public void testUserHasVoted() throws IOException {
+    when(mockedRequest.getParameter(MeetingTimeFields.MEETING_TIME_ID)).thenReturn(MEETING_TIME_ID_VAL); 
+    when(mockedRequest.getParameter(MeetingTimeFields.VOTERS)).thenReturn(EXISTING_VOTER);
+    new VoteMeetingTimeServlet().doPost(mockedRequest, mockedResponse);
+    
+    // check that error response was sent with appropriate details
+    testBadRequest(HttpServletResponse.SC_CONFLICT, ErrorMessages.USER_HAS_VOTED_ERROR);
+  }
+
+  /** Voter is the first voter - there were no past voters for the meeting time */
+  @Test
+  public void testFirstVoter() throws IOException {
+    // set the voters list in the hardcoded entity to be an empty list
+    Entity meetingTime = getHardcodedEntity();
+    meetingTime.setProperty(MeetingTimeFields.VOTERS, new ArrayList<String>());
+    // set the vote count to be 0
+    meetingTime.setProperty(MeetingTimeFields.VOTE_COUNT, 0);
+    datastore.put(meetingTime);
+
+    when(mockedRequest.getParameter(MeetingTimeFields.MEETING_TIME_ID)).thenReturn(MEETING_TIME_ID_VAL); 
+    when(mockedRequest.getParameter(MeetingTimeFields.VOTERS)).thenReturn(NEW_VOTER);
+    new VoteMeetingTimeServlet().doPost(mockedRequest, mockedResponse);
+
+    // Check that the entity was updated as expected.
+    Entity updatedMeetingTime = getHardcodedEntity();
+
+    // voters should now JUST include NEW_VOTER, and NOT the original voters
+    HashSet<String> voters = new HashSet<String>( (ArrayList<String>) updatedMeetingTime.getProperty(MeetingTimeFields.VOTERS));
```

Comment:
nit: please fix the formatting of this line (e.g. line too long, space between 2 open open parenthesises)